sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java

```diff
@@ -151,6 +151,21 @@ public void disablePersisted() {
         verify(mChannel, never()).addListener(any(Channel.Listener.class));
     }
 
+    @Test
+    public void listWhenOffline() {
+        when(mNetworkStateHelper.isNetworkConnected()).thenReturn(false);
+
+        /* Make the call. */
+        PaginatedDocuments<TestDocument> docs = Data.list(USER_DOCUMENTS, TestDocument.class).get();
+
+        /* Verify the result correct. */
+        assertFalse(docs.hasNextPage());
+        Page<TestDocument> page = docs.getCurrentPage();
+        assertNull(page.getItems());
+        assertNotNull(page.getError());
+        assertEquals(DataException.class, page.getError().getClass());
+    }
```

Comment:
In addition, you can add verification that it doesn't try to make HTTP calls

Reply:
Good point!